File: src/types/assistant.ts
export type ChatRole = 'user' | 'assistant';

export interface ChatMessage {
  id: string;
  role: ChatRole;
  content: string;
  /** Which real backend tools the assistant called to ground this reply — empty for general-knowledge replies. */
  toolsCalled: string[];
  createdAt: string;
}

File: backend/app/schemas/assistant.py
from datetime import datetime
from typing import Literal

from app.schemas.base import CamelModel

ChatRole = Literal["user", "assistant"]


class ChatMessageOut(CamelModel):
    id: str
    role: ChatRole
    content: str
    tools_called: list[str]
    created_at: datetime


class SendMessageRequest(CamelModel):
    message: str
